cel/test/wait.go

```diff
@@ -0,0 +1,103 @@
+/*
```

Comment:
This file looks largely lifted from https://github.com/tektoncd/pipeline/blob/main/test/wait.go. This does seem like useful functionality we'd want to reuse across tests. I wonder if it would make sense to just depend on https://pkg.go.dev/github.com/tektoncd/pipeline/test directly instead of forking these.

Reply:
added the missing function for polling for runs into that file in pipelines so that we can reuse it across different custom tasks -- https://github.com/tektoncd/pipeline/pull/4213